orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -412,8 +410,12 @@ def _build_training_graph(self, *inputs, **kwargs):
         self._onnx_graphs_info = self._module_gradient_graph_builder.get_training_graph_info()
 
         if self._save_onnx:
-            onnx.save(self._onnx_training,
-                      self._save_onnx_prefix + '_training.onnx')
+            inference_optimized_model = onnx.load_model_from_string(
+                self._module_gradient_graph_builder.get_inference_optimized_model())
+
+            onnx.save(self._onnx_inference, self._save_onnx_prefix + '_inference.onnx')
```

Comment:
The reason the inference save was above was that sometimes we can export the inference graph, but not the training. If we leave everything to the end, we dont get the inference one

Reply:
good point.